Marlin/buzzer.h

```diff
@@ -20,11 +20,67 @@
  *
  */
 
-#ifndef BUZZER_H
-  #define BUZZER_H
+#ifndef __BUZZER_H__
+#define __BUZZER_H__
 
-  #if HAS_BUZZER
-    void buzz(long duration, uint16_t freq);
-  #endif
+#include "Marlin.h"
+#include "circularqueue.h"
 
-#endif //BUZZER_H
+#define TONE_QUEUE_LENGTH 8
+
+struct tone_t {
+  uint16_t duration;
+  uint16_t frequency;
+};
+
+class Buzzer {
```

Comment:
Scott I believe we have to check if our definition of `Singleton` is the same.. so far I don't believe we have a single `Singleton` (excuse the word redundancy) on Marlin.. we have classes which we, by a code of honor, only instantiate once.

Concerning the static, have a look https://github.com/MarlinFirmware/Marlin/issues/3680#issuecomment-220385645.